docs/architecture.md

```diff
@@ -4,38 +4,61 @@ title: Architecture
 hide_title: false
 ---
 
+## Architecture
+
 Below, find a diagram with the components that comprise an instance of Cumulus.
 
 ![Architecture diagram of a Cumulus deployment](assets/cumulus-arch-diagram.png)
 
-While the internals are complex, the boxes represent components which are easy to understand:
+This diagram details all of the major architectural  components of a Cumulus deployment.
+
+While the diagram can feel complex, it can easily be digested in several major components:
+
+### Data Distribution
+
+End Users can access data via Cumulus's `distribution` submodule, which includes ASF's [thin egress application](https://github.com/asfadmin/thin-egress-app), this provides authenticated data egress, temporary S3 links and other statistics features.
+
+#### Data search
+
+End user exporsure of Cumulus's holdings is expected to be provided by an external service.
+
+For NASA use, this is assumed to be [CMR](<https://earthdata.nasa.gov/eosdis/science-system-description/eosdis-components/cmr>) in this diagram.
+
+### Data ingest
+
+#### Workflows
+
+The core of the ingest and processing capabilities in Cumulus is built into the deployed AWS [Step Function](https://aws.amazon.com/step-functions/) workflows.    Cumulus rules trigger workflows via either Cloud Watch rules, Kinesis streams, SNS topic, or SQS queue.   The workflows then run with a configured [Cumulus message](./workflows/cumulus-task-message-flow), utilizing built-in processes to report status of granules, PDRs, executions, etc to the [Data Persistence](#data-persistence) components.
+
+Workflows can optionally report granule metadata to [CMR](<https://earthdata.nasa.gov/eosdis/science-system-description/eosdis-components/cmr>), and if configured to use the [Cumulus message adapter (CMA)](<https://github.com/nasa/cumulus-message-adapter>), workflow steps can report metrics information to a shared SNS topic, which could be subscribed to for near real time granule, execution, and PDR status. This could be used for metrics reporting using an external ELK stack, for example.
```

Comment:
```suggestion
Workflows can optionally report granule metadata to [CMR](<https://earthdata.nasa.gov/eosdis/science-system-description/eosdis-components/cmr>), and workflow steps can report metrics information to a shared SNS topic, which could be subscribed to for near real time granule, execution, and PDR status. This could be used for metrics reporting using an external ELK stack, for example.
```